### PR TITLE
feat: Add customization to add all person profile properties as setPersonPropertiesForFlags

### PR DESCRIPTION
**Pull request by robbie-c**
## Changes
Add a customization to set all known person properties with setPersonPropertiesForFlags

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers
- [x] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { version } from '../package.json'
+import packageInfo from '../package.json'
```
Comment (robbie-c) [inline src/config.ts:1]: I got a warning about this, so I fixed it as a drive-by

`Should not import the named export 'version' (imported as 'version') from default-exporting module (only default export is available soon)`

File: src/customizations/setAllPersonProfilePropertiesAsPersonPropertiesForFlags.test.ts
[the remark this answers is not in the document]
Comment (robbie-c) [reply src/customizations/setAllPersonProfilePropertiesAsPersonPropertiesForFlags.test.ts:1]: Ah I'll move it

File: src/utils/event-utils.ts [hunk 2]
```diff
@@ -31,6 +31,25 @@ export const CAMPAIGN_PARAMS = [
     'rdt_cid', // reddit
 ]
 
+export const EVENT_TO_PERSON_PROPERTIES = [
+    // mobile params
+    '$app_build',
+    '$app_name',
+    '$app_namespace',
+    '$app_version',
```
Comment (marandaneto) [inline src/utils/event-utils.ts:39]: There are many more params than that such as `$device_manufacturer`, `$device_model`, etc, I wonder if this list is exhaustive to only what this SDK specifically collects?

File: src/utils/event-utils.ts [hunk 3]
```diff
@@ -31,6 +31,25 @@ export const CAMPAIGN_PARAMS = [
     'rdt_cid', // reddit
 ]
 
+export const EVENT_TO_PERSON_PROPERTIES = [
```
Comment (pauldambra) [inline src/utils/event-utils.ts:34]: I think this could do with a comment or a better name... 

I think it's saying "these are event properties that we will copy to person properties"
but I don't know for sure 🙈

File: src/customizations/setAllPersonProfilePropertiesAsPersonPropertiesForFlags.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { PostHog } from '../posthog-core'
+import { CAMPAIGN_PARAMS, EVENT_TO_PERSON_PROPERTIES, Info } from '../utils/event-utils'
+import { each, extend, includes } from '../utils'
+
+export const setAllPersonProfilePropertiesAsPersonPropertiesForFlags = (posthog: PostHog): void => {
```
Comment (pauldambra) [inline src/customizations/setAllPersonProfilePropertiesAsPersonPropertiesForFlags.ts:5]: could also do with a comment IMHO, why would I want to do this?

**Review by pauldambra — APPROVED** [review]
👍 

couple of comments but nothing blocking

File: src/utils/event-utils.ts (hunk 3 above)
Comment (robbie-c) [reply src/utils/event-utils.ts:34]: I don't disagree but it's copied directly from plugin server with that name, I'll add a comment though!

File: src/utils/event-utils.ts (hunk 2 above)
Comment (robbie-c) [reply src/utils/event-utils.ts:39]: Those other ones aren't converted into person properties though, or are they? This list comes straight from plugin-server.

**Comment by robbie-c** [comment]
I'm going to merge this now and add a couple of comments in a follow up, just as I'm in a few meetings this morning and it'd be good to get back to the customer asap

File: src/utils/event-utils.ts (hunk 2 above)
Comment (marandaneto) [reply src/utils/event-utils.ts:39]: I didn't check so the plugin-server should be up to date and all good.
